### PR TITLE
gui widgets layout

### PR DESCRIPTION
**Pull request by liibin**
After opening a project, it looks like this.
image

### CODE REVIEW

File: src/MainWindow/main_window.cpp [hunk 1]
```diff
@@ -181,21 +189,30 @@ void MainWindow::ReShowWindow(QString strProject) {
...
-  QDockWidget* editorDockWidget = new QDockWidget(this);
-  editorDockWidget->setObjectName("editordockwidget");
+  // QDockWidget* editorDockWidget = new QDockWidget(this);
+  // editorDockWidget->setObjectName("editordockwidget");
   TextEditor* textEditor = new TextEditor(this);
   textEditor->RegisterCommands(GlobalSession);
   textEditor->setObjectName("textEditor");
-  editorDockWidget->setWidget(textEditor->GetTextEditor());
-  addDockWidget(Qt::RightDockWidgetArea, editorDockWidget);
+  // editorDockWidget->setWidget(textEditor->GetTextEditor());
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:199]: please remove all commented code

File: src/MainWindow/main_window.cpp [hunk 2]
```diff
@@ -167,7 +175,7 @@ void MainWindow::createActions() {
   });
 }
 
-void MainWindow::gui_start() { ReShowWindow("unknown"); }
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:170]: The idea of this change was to show everything at the start, all widget. Even if no project started.

File: src/MainWindow/main_window.cpp [hunk 3]
```diff
@@ -181,21 +189,25 @@ void MainWindow::ReShowWindow(QString strProject) {
...
   connect(textEditor, SIGNAL(CurrentFileChanged(QString)), sourForm,
           SLOT(SetCurrentFileItem(QString)));
 
+  QWidget* centralWidget = new QWidget(this);
+  QBoxLayout* layout = new QBoxLayout(QBoxLayout::TopToBottom, centralWidget);
+  layout->setContentsMargins(0, 0, 0, 0);
+  layout->setSpacing(0);
+  layout->addWidget(textEditor->GetTextEditor());
+  centralWidget->setLayout(layout);
+  takeCentralWidget();
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:208]: it is doing nothing

Comment (liibin) [reply src/MainWindow/main_window.cpp:208]: the next line is
setCentralWidget(centralWidget);

Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp:208]: so why do you need to call takeCentralWidget(); ?

File: src/MainWindow/main_window.cpp (hunk 1 above)
Comment (liibin) [reply src/MainWindow/main_window.cpp]: OK.

File: src/MainWindow/main_window.cpp (hunk 2 above)
Comment (liibin) [reply src/MainWindow/main_window.cpp]: > The idea of this change was to show everything at the start, all widget. Even if no project started.

I'll change it back.

**Comment by KochynVolodymyr** [comment]
Close #212

**Review by KochynVolodymyr — APPROVED** [review]

**Pull request merged**